src/Serialization/main.jl

```diff
@@ -53,16 +54,34 @@ end
 # registerSerializationType is a macro to ensure that the string we generate
 # matches exactly the expression passed as first argument, and does not change
 # in unexpected ways when import/export statements are adjusted.
+# It also sets the value of serialize_with_id, which determines
```

Comment:
Actually it is more complex now, because also `has_elem_basic_encoding` plays a role... 

I am wondering now if we should rethink the relation between `has_elem_basic_encoding` and `serialize_with_id` and this macro... Like, when I specify that "elements have basic encoding", then we automatically don't store the id. Only if element do *not* have basic encoding is it even an option to store the id. Perhaps we can come up with a slightly less complex way to specify all of this?

Reply:
I've opted to keep these two ideas seperate to avoid confusion or possible inconsistences in the future. i.e. Objects whose elements have basic encodings may also be stored with an id. Having elements with basic encoding just means that the object has no base_ring, and represents a sort of atomic fundamental type at least within oscar.

Reply:
I've also removed the dependancy to the has_elem_basic_encoding